### PR TITLE
Implement return values for enhanced PutItem

### PR DESCRIPTION
**Pull request by dagnir**
## Motivation and Context
Requested feature.

## Description
This commit introduces DynamoDbTable#putItemWithResponse() that allows customers
to specify additional parameters on the request such as ReturnValues to get
additional information the service response.

Relates to #1918

## Testing
- Unit tests
- Integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by dagnir** [comment]
The flagged bug is a false positive:

```java
        if (request.left().isPresent()) {
            requestBuilder = addPlainPutItemParameters(requestBuilder, request.left().get());
```

It says 

> Call "Optional#isPresent()" before accessing the value. 

But we are doing that.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/PutItemOperation.java [hunk 1]
```diff
@@ -99,18 +101,31 @@ public PutItemRequest generateRequest(TableSchema<T> tableSchema,
                                                               .tableName(operationContext.tableName())
                                                               .item(itemMap);
 
+        if (request.left().isPresent()) {
+            requestBuilder = addPlainPutItemParameters(requestBuilder, request.left().get());
+        }
```
Comment (millems) [inline services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/PutItemOperation.java:106]: Nit:
```java
request.left().ifPresent(r -> addPlainPutItemParameters(requestBuilder, r));
```

**Review by millems — APPROVED** [review]
Just one minor comment

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/PutItemOperation.java (hunk 1 above)
Comment (dagnir) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/PutItemOperation.java:106]: Tried it originally, but java complains that `requestBuilder` isn't effectively final ☹️

**Pull request merged**